src/Microsoft.AspNet.Hosting.Server.Abstractions/IHttpApplication.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Hosting.Server
+{
+    /// <summary>
+    /// Represents an HttpApplication.
+    /// </summary>
+    public interface IHttpApplication
+    {
+        /// <summary>
+        /// Create an HttpContext given a collection of HTTP features.
+        /// </summary>
+        /// <param name="contextFeatures">A collection of HTTP features to be used for creating the HttpContext.</param>
+        /// <returns>The created HttpContext.</returns>
+        object CreateHttpContext(object contextFeatures);
```

Comment:
But that will add a dependency on `Microsoft.AspNet.Http.Features` which I thought we were going to merge into abstractions at some point?